interface/ceed.c

```diff
@@ -475,16 +474,17 @@ int CeedSetBackendFunction(Ceed ceed,
   char lookupname[CEED_MAX_RESOURCE_LEN+1] = "";
 
   // Build lookup name
+  if (strcmp(type, "Ceed"))
+    strncat (lookupname, "Ceed", CEED_MAX_RESOURCE_LEN);
```

Comment:
What is this case handling?

Reply:
This keeps the backend code looking a little nice.

We use
```
  ierr = CeedSetBackendFunction(ceed, "Vector", vec, "SetArray",
                                CeedVectorSetArray_Ref); CeedChk(ierr);
  ierr = CeedSetBackendFunction(ceed, "Vector", vec, "GetArray",
                                CeedVectorGetArray_Ref); CeedChk(ierr);
  ierr = CeedSetBackendFunction(ceed, "Vector", vec, "GetArrayRead",
                                CeedVectorGetArrayRead_Ref); CeedChk(ierr);
  ierr = CeedSetBackendFunction(ceed, "Vector", vec, "RestoreArray",
                                CeedVectorRestoreArray_Ref); CeedChk(ierr);
  ierr = CeedSetBackendFunction(ceed, "Vector", vec, "RestoreArrayRead",
                                CeedVectorRestoreArrayRead_Ref); CeedChk(ierr);
  ierr = CeedSetBackendFunction(ceed, "Vector", vec, "Destroy",
                                CeedVectorDestroy_Ref); CeedChk(ierr);
```

instead of
```
  ierr = CeedSetBackendFunction(ceed, "CeedVector", vec, "SetArray",
                                CeedVectorSetArray_Ref); CeedChk(ierr);
  ierr = CeedSetBackendFunction(ceed, "CeedVector", vec, "GetArray",
                                CeedVectorGetArray_Ref); CeedChk(ierr);
  ierr = CeedSetBackendFunction(ceed, "CeedVector", vec, "GetArrayRead",
                                CeedVectorGetArrayRead_Ref); CeedChk(ierr);
  ierr = CeedSetBackendFunction(ceed, "CeedVector", vec, "RestoreArray",
                                CeedVectorRestoreArray_Ref); CeedChk(ierr);
  ierr = CeedSetBackendFunction(ceed, "CeedVector", vec, "RestoreArrayRead",
                                CeedVectorRestoreArrayRead_Ref); CeedChk(ierr);
  ierr = CeedSetBackendFunction(ceed, "CeedVector", vec, "Destroy",
                                CeedVectorDestroy_Ref); CeedChk(ierr);
```

and so on for all objects.

Reply:
@jedbrown should I change it or are we good to merge as is? I don't feel strongly either way, as its just minor cosmetics.

Reply:
Any preference @jedbrown? This is the last thing to clear up before merging.

Reply:
I'm okay either way so long as it's documented.

Reply:
```
/**
  @brief Set a backend function

  This function is used for a backend to set the function associated with
  the CEED objects. For example,
    CeedSetBackendFunction(ceed, "Ceed", ceed, "VecCreate", BackendVecCreate)
  sets the backend implementation of 'CeedVectorCreate' and
    CeedSetBackendFunction(ceed, "Basis", basis, "Apply", BackendBasisApply)
  sets the backend implementation of 'CeedBasisApply'. Note, the prefix 'Ceed'
  is not required for the object type ("Basis" vs "CeedBasis").

  @param ceed           Ceed for error handling
  @param type           Type of Ceed object to set function for
  @param[out] object    Ceed object to set function for
  @param fname          Name of function to set
  @param f              Function to set

  @return An error code: 0 - success, otherwise - failure

  @ref Advanced
**/
```

Updated the documentation. In this case, "CeedBasis" and "Basis" both work, which is kind of nice.

Reply:
LGTM.  Note that `VecCreate` should really be called `VectorCreate` (no ad-hoc abbreviations; that was mistakenly missed in 4ff0612a4ce113899b035b1e6340babb64a59988).

Reply:
It fits in the theme of this PR, so I'll fix it here.